pkg/kubelet/prober/manager.go

```diff
@@ -207,6 +207,23 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *api.PodStatus) {
 		}
 		podStatus.ContainerStatuses[i].Ready = ready
 	}
+	// init containers are ready if they have exited with success or if a readiness probe has
+	// succeeded.
+	for i, c := range podStatus.InitContainerStatuses {
```

Comment:
Remove dead code? Or is it actually required for something higher up the stack to not barf, if so  suggest a TODO. 

It looks like the proposal leaves readiness as an open question, so leaving it in is confusing, especially if it needs to be preserved across a refactor.